### PR TITLE
feat: convert cobra commands from Run to RunE

### PR DESCRIPTION
**Pull request by radekska**
this PR solves https://github.com/alegrey91/fwdctl/issues/110 by converting cobra commands to use `RunE` instead of `Run`

### CODE REVIEW

File: cmd/apply.go [hunk 1]
```diff
@@ -32,61 +33,46 @@ var applyCmd = &cobra.Command{
...
-			// add slot to buffered channel
-			chLimit <- 1
-			go func(rule iptables.Rule, wg *sync.WaitGroup, chErr chan error, chLimit chan int) {
-				err := ipt.ValidateForward(&rule)
-				wg.Done()
-				chErr <- err
-				// free slot from buffered channel
-				<-chLimit
-			}(rule, &wg, chErr, chLimit)
+			r := &rule
```
Comment (alegrey91) [inline cmd/apply.go:54]: can you explain why you want to change this?

**Review by alegrey91 — COMMENTED** [review]
looks good, just want to know more about why you want to use `errgroup` instead of current implementation.

**Comment by radekska** [comment]
> looks good, just want to know more about why you want to use errgroup instead of current implementation.

`errgroup` is a package that allows you to synchronize multiple goroutines similar to `sync.WaitGroup` but with the addition of propagating errors to a caller without using extra channels, which is exactly what is needed in this case.

File: cmd/apply.go (hunk 1 above)
Comment (radekska) [reply cmd/apply.go:54]: TLDR; when running a goroutine with function closure (`rule` variable) there is for each of them the last element of `ruleSet.Rules` will be used.
Creating a copy (`r := &rule`) ensures that each goroutine processes unique elements.
https://go.dev/wiki/CommonMistakes#using-goroutines-on-loop-iterator-variables

Comment (alegrey91) [reply cmd/apply.go:54]: Ok, got it. That's fine for me, just add a `g.SetLimit(10)` before the `g.Go()` function to limit the number of active goroutines as was previously implemented with `WaitGroup`.

**Comment by alegrey91** [comment]
thanks @radekska for the contribution.
Really appreciated :)

**Comment by radekska** [comment]
> thanks @radekska for the contribution. Really appreciated :)

glad I could help!

**Pull request merged**